src/main/java/com/carpool/car_pool/services/RideOfferService.java

```diff
@@ -91,4 +90,15 @@ public RideOfferResponse editRideOfferDetail(EditRideOfferRequest editRideofferR
 
 
     }
+
+    public void deleteRideOffer(long id, UserEntity user) {
+        RideOfferEntity rideOffer = rideOfferRepository.findById(id)
+                .orElseThrow(()-> new RuntimeException("Ride offer does not exist"));
+
+        if (!rideOffer.getCreator().equals(user)){
+            throw new RuntimeException("Only owner can delete this ride offer");
+        }
+
+        rideOfferRepository.delete(rideOffer);
+    }
```

Comment:
It would be good to return ID or something for us for easier testing with postman and posibility of displaying that to the user. For example we could show to the user message "Ride {id} has been deleted"